### PR TITLE
Fix some of the typing around the product-entity reducer

### PR DESCRIPTION
**Pull request by lderrickable**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?
```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
The type for a product entity was still a DaffProduct, while it should be a DaffProductUnion.

Fixes: N/A

## What is the new behavior?
The type for a product entity is a DaffProductUnion and a factory was added for DaffProductUnions

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: libs/product/src/reducers/index.spec.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@ import { DaffProduct } from "../models/product";
 describe('selectProductState', () => {
 
   let store: Store<fromProduct.State>;
-  const productFactory: DaffProductFactory = new DaffProductFactory();
+  const productUnionFactory: DaffProductUnionFactory = new DaffProductUnionFactory();
```
Comment (damienwebdev) [inline libs/product/src/reducers/index.spec.ts]: Since the `union` is so broad, can we get away with only using the `DaffProductFactory` here?

Comment (lderrickable) [reply libs/product/src/reducers/index.spec.ts]: So we can get rid of the union factory entirely?

Comment (damienwebdev) [reply libs/product/src/reducers/index.spec.ts]: Yes.

Comment (lderrickable) [reply libs/product/src/reducers/index.spec.ts]: removed ProductUnion factory

**Pull request merged**